packages/dd-trace/test/appsec/waf/waf_context_wrapper.spec.js

```diff
@@ -27,14 +41,45 @@ describe('WAFContextWrapper', () => {
     wafContextWrapper.run(payload)
 
     expect(ddwafContext.run).to.have.been.calledOnceWithExactly(payload, 1000)
+    expect(reportMetricsStub).to.have.been.calledOnce
   })
 
-  it('Should send ephemeral addresses every time', () => {
+  it('Should send HTTP_INCOMING_QUERY twice if waf run returns null', () => {
     const ddwafContext = {
       run: sinon.stub()
     }
     const wafContextWrapper = new WAFContextWrapper(ddwafContext, 1000, '1.14.0', '1.8.0', knownAddresses)
 
+    const payload = {
+      persistent: {
+        [addresses.HTTP_INCOMING_QUERY]: { key: 'value' }
+      }
+    }
+
+    wafContextWrapper.run(payload)
+    wafContextWrapper.run(payload)
+
+    expect(ddwafContext.run).to.have.been.calledTwice
+    expect(ddwafContext.run).to.have.been.calledWithExactly(payload, 1000)
```

Comment:
The test title says you're making sure it's sent twice, then why don't you actually check it's sent twice
```suggestion
    expect(ddwafContext.run.firstCall).to.have.been.calledWithExactly(payload, 1000)
    expect(ddwafContext.run.secondCall).to.have.been.calledWithExactly(payload, 1000)
```

Reply:
I do check `reportMetrics` instead

Reply:
yes but here the test is to make sure `addressesToSkip` is not updated after a failed waf call, metrics have nothing to do with this, so you need to make sure the waf is called twice with the address, to make sure `addressesToSkip` wasn't updated in the firsrt call

Reply:
I thought this was already what you meant by writing this test, otherwise i have no idea why you called it "should call querystring twice"

Reply:
I kept the title and check `ddwafContext.run` calls, it's not related to my PR but nice to have it